modules/ztp-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,296 @@
+type: Opaque
+data:
+  seedAuth: <encoded_AUTHFILE> <2>
+  hubKubeconfig: <encoded_KUBECONFIG> <3>
+----
+<1> The `Secret` CR must have the `name: seedgen` and `namespace: openshift-lifecycle-agent` fields.
+<2> Specifies a base64-encoded authfile for write-access to the registry for pushing the generated seed images.
+<3> Optional. Specifies a base64-encoded `KUBECONFIG` file for administrator access to the hub cluster to unregister the seed cluster from {rh-rhacm}. If this is not present in the secret, the {rh-rhacm} cleanup is skipped.
+--
+
+.. Create the authentication and the `Kubeconfig` files by running the following commands:
++
+--
+.Authentication file
+[source,terminal]
+----
+$ MY_USER=myuserid
+$ AUTHFILE=/tmp/my-auth.json
+$ podman login --authfile ${AUTHFILE} -u ${MY_USER} quay.io/${MY_USER}
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${AUTHFILE} ; echo
+----
+
+.hubKubeconfig file
+[source,terminal]
+----
+$ HUBKUBECONFIGFILE=/home/<user_ID>/hubkubeconfig
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${HUBKUBECONFIGFILE} ; echo
+----
+--
+
+.. Copy the outputs into the respective `seedAuth` and `hubKubeconfig` fields in the `Secret` CR.
+
+. Apply the `Secret` CR.
++
+[source,terminal]
+----
+$ oc apply -f secretseedgenerator.yaml
+----
+
+. (Optional) If you did not populate the `hubKubeconfig` field in the `Secret` CR, detach the seed cluster from the hub cluster and remove all {rh-rhacm} and multicluster engine certificates that cannot be recertificated in the target cluster.
+
+.. Detach the seed cluster from the hub cluster:
++
+[source,terminal]
+----
+$ oc delete managedcluster sno-worker-00
+----
+
+.. Wait until the `ManagedCluster` CR is removed. Once the CR is removed and you create the proper `SeedGenerator` CR, the {lcao} cleans up the {rh-rhacm} artifacts.
+
+. Create the `SeedGenerator` CR:
++
+--
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage <1>
+spec:
+  seedImage: <seed_container_image> <2>
+#  recertImage: <recertification_tool_container_image> <3>
+----
+<1> The `SeedGenerator` CR must be named `seedimage`.
+<2> Specify the container image URL, for example, `quay.io/example/seed-container-image`.
+//<3> Optional. Specify the recertification tool container image to pass to the {lcao}, for example: `quay.io/example/recert`.
+
+
+[IMPORTANT]
+====
+The same version of the applications must function on both the current and the target release of {product-title}.
+====
+--
+
+. Generate the seed image by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f seedgenerator.yaml
+----
+
++
+[IMPORTANT]
+====
+The cluster loses API capabilities while the {lcao} generates the seed image.
+====
+
++
+Applying the `SeedGenerator` CR stops the `kubelet` and the CRI-O operations, and then it starts the image generation.
+Finally, the {lcao} starts a cleanup process that allows you to keep the seed {sno} fully functional.
+
```

Comment:
As stated in a previous comment, yes, the seed cluster is fully functional again. But it is not supported to upgrade the seed cluster and run another seed image creation. User must provision a new seed cluster with the proper version they want to create the seed image.